extractors/mongodb/extractor.go

```diff
@@ -0,0 +1,118 @@
+package mongodb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type Extractor struct{}
+
+type Result struct {
+	CollectionName string
+	DatabaseName   string
+	Indexes        []bson.D
+}
+
+type Config struct {
+	UserID   string `mapstructure:"user_id"`
+	Password string `mapstructure:"password"`
+	Host     string `mapstructure:"host"`
+}
+
+func (e *Extractor) Extract(configMap map[string]interface{}) (result []Result, err error) {
+	config, err := e.getConfig(configMap)
+	if err != nil {
+		return
+	}
+	err = e.validateConfig(config)
+	if err != nil {
+		return
+	}
+	uri := "mongodb://" + config.UserID + ":" + config.Password + "@" + config.Host
+	clientOptions := options.Client().ApplyURI(uri)
+	result, err = ListCollections(clientOptions)
+	if err != nil {
+		return
+	}
+	fmt.Println(result)
+	return result, err
+}
+
+func ListCollections(clientOptions *options.ClientOptions) (result []Result, err error) {
```

Comment:
should we change this to private function?
wdyt?

Reply:
ok, so I make these methods private, since we are not using them outside the packages.